HW1/P2/P2b.py

```diff
@@ -1,3 +1,30 @@
 from P2 import *
+import random
 
-# Your code here
+
+def compute_image_balanced(s, p):
+    """Compute Mandelbrot image using balanced partitioning strategy.
+    :param s: The size of the square image to compute.
+    :param p: The square root of the number of partitions.
+    """
+    c = s / 4.0
+
+    ij = sc.parallelize([(ind / s, ind % s) for ind in sorted(
+        xrange(s * s), key=lambda k: random.random())], p * p)
```

Comment:
This basically creates the ranges is python, and only then parallelizes them using spark.
You should have used sc.range for that.